cidr/tree4.go

```diff
@@ -74,9 +81,10 @@ func (tree *Tree4) AddCIDR(cidr *net.IPNet, val interface{}) {
 
 	// Final node marks our cidr, set the value
 	node.value = val
+	tree.list = append(tree.list, entry{CIDR: cidr, Value: &val})
```

Comment:
Do we need to update the `tree.list` entry earlier for the "We already have this range so update the value" short circuit?

Reply:
Nice catch and yes.